src/lib.rs

```diff
@@ -172,7 +196,7 @@ impl Multihash {
             });
         }
         Ok(Multihash {
-            bytes: Bytes::from(bytes),
+            storage: Storage::from_slice(&bytes),
```

Comment:
Like above, why not keep the existing `Vec` and use `Storage::Heap(bytes.into())`?

Reply:
If the vec contains <= MAX_INLINE bytes, you don't want Heap.

For the case where inline storage is not possible, do you think that .into() is more efficient to go from a `Vec<u8>` to an `Arc<[u8]>`?

Reply:
Looking at the From instance that does the conversion, going from a Vec<u8> to a Arc<[u8]> does not seem to gain you anything compared to just copying from a slice.

https://doc.rust-lang.org/src/alloc/sync.rs.html#2021-2033